pandas/core/reshape/pivot.py

```diff
@@ -454,6 +454,8 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
 
     from pandas import DataFrame
     df = DataFrame(data, index=common_idx)
+    if not len(df):
+        return DataFrame()
     if values is None:
```

Comment:
pls add a test for this case. I don't think we have coverage as your change didn't break things (but should have). (I think this was a crosstabe of empty frames?)

Reply:
great. the key is that the index IS preserved.

Reply:
sorry, I mean that the test this was hitting was maybe not right, your change here *should* have broken it